include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +513,58 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class RandomAccessIterator>
+auto mode(RandomAccessIterator first, RandomAccessIterator last)
+{
+    using Real = typename std::iterator_traits<RandomAccessIterator>::value_type;
+    using Size = typename std::iterator_traits<RandomAccessIterator>::difference_type;
+
+    std::vector<Real> sorted(first, last);
+    std::sort(sorted.begin(), sorted.end());
+
+    std::vector<Real> modes {};
+    Size max_counter {};
+    Real previous_search {boost::math::tools::max_value<Real>()};
+
+    if(sorted.front() == sorted.back())
+    {
+        modes.push_back(sorted.front());
+        return modes;
+    }
+
+    for(const Real i : sorted)
+    {
+        if(i == previous_search)
+        {
+            continue;
+        }
+        
+        Size current_count {std::count(sorted.begin(), sorted.end(), i)};
```

Comment:
The complexity of `std::count` is linear, so isn't this loop quadratic?

Reply:
It would be. I swapped the range based for loop out with an iterator so it can replace begin() in the count. That will reduce the range it is iterating over.